src/pt/mangaterra/src/eu/kanade.tachiyomi.extension.pt.mangaterra/MangaTerra.kt

```diff
@@ -0,0 +1,235 @@
+        val mangaChapterUrl = document.location()
+        val maxPage = findPageCount(mangaChapterUrl)
+        return (1..maxPage).map { page -> Page(page - 1, "$mangaChapterUrl/$page") }
+    }
+
+    override fun popularMangaFromElement(element: Element) = SManga.create().apply {
+        title = element.selectFirst("p")!!.ownText()
+        thumbnail_url = element.selectFirst("img")?.srcAttr()
+        setUrlWithoutDomain(element.selectFirst("a")!!.absUrl("href"))
+    }
+
+    override fun popularMangaNextPageSelector() = ".pagination > .page-item:not(.disabled):last-child"
+
+    override fun popularMangaRequest(page: Int) = GET("$baseUrl/manga?q=p&page=$page", headers)
+
+    override fun popularMangaSelector(): String = ".card-body .row > div"
+
+    override fun searchMangaFromElement(element: Element) = popularMangaFromElement(element)
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        if (response.request.url.pathSegments.contains("search")) {
+            return searchByQueryMangaParse(response)
+        }
+        return super.searchMangaParse(response)
+    }
+
+    override fun searchMangaNextPageSelector() = popularMangaNextPageSelector()
+
+    override fun fetchSearchManga(page: Int, query: String, filters: FilterList): Observable<MangasPage> {
+        if (query.startsWith(slugPrefix)) {
+            val slug = query.substringAfter(slugPrefix)
+            return client.newCall(GET("$baseUrl/manga/$slug", headers))
+                .asObservableSuccess().map { response ->
+                    MangasPage(listOf(mangaDetailsParse(response)), false)
+                }
+        }
+        return super.fetchSearchManga(page, query, filters)
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = baseUrl.toHttpUrl().newBuilder()
+
+        if (query.isNotBlank()) {
+            url.addPathSegment("search")
+                .addQueryParameter("q", query)
+            return GET(url.build(), headers)
+        }
+
+        url.addPathSegment("manga")
+
+        filters.forEach { filter ->
+            when (filter) {
+                is GenreFilter -> {
+                    filter.state.forEach {
+                        if (it.state) {
+                            url.addQueryParameter(it.query, it.value)
+                        }
+                    }
+                }
+                else -> {}
+            }
+        }
+
+        url.addQueryParameter("page", "$page")
+
+        return GET(url.build(), headers)
+    }
+
+    override fun searchMangaSelector() = popularMangaSelector()
+
+    override fun getFilterList(): FilterList {
+        CoroutineScope(Dispatchers.IO).launch { fetchGenres() }
+        val filters = mutableListOf(
+            GenreFilter(
+                title = "Categorias",
+                genres = genresList,
+            ),
+        )
+        return FilterList(filters)
+    }
```

Comment:
add a warning/note to tell user to press reset to load the genrelist. also don't add when it is empty